constraintlayout/constraintlayout/src/main/java/androidx/constraintlayout/widget/ConstraintSet.java

```diff
@@ -1106,6 +1107,48 @@ static void parseDimensionConstraintsString(Object data, String value, int orien
         }
     }
 
+
+    /**
+     * Get the types associated with this ConstraintSet
+     * The types mechanism allows you to tag the constraint set
+     * with a series of string to define properties of a ConstraintSet
+     *
+     * @return an array of type strings
+     */
+    public String[] getConstraintSetTypes() {
```

Comment:
Tags will be confused with view.setTage view getTag
So can't use tags

Reply:
After our conversation, this is what Imagine it would look like:
```Java
class ConstraintSet {
    public void setMatchingLabels(String... labels)

    public String[] getMatchingLabels()
}

class MotionLayout {
   public int[] getMatchingConstraintSetIds(String... labels)
}

<ConstraintSet
  motion:constraintSetMatchingLabels="a,b,1,2"
/>
```